testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -1903,6 +1903,47 @@ public void testDeleteAfterMajorCompaction()
         });
     }
 
+    @Test
+    public void testInsertIntoUnbucketedTransactionalTableWithTaskWriterCounterGE1()
+    {
+        withTemporaryTable("test_insert_into_unbucketed_transactional_table", true, false, NONE, tableName -> {
+            onTrino().executeQuery(format("CREATE TABLE %s  WITH (format='ORC', transactional=true) AS TABLE tpch.tiny.nation WITH NO DATA;", tableName));
+            onTrino().executeQuery("SET SESSION task_writer_count = 2");
+            onTrino().executeQuery(format("INSERT INTO %s SELECT * FROM tpch.tiny.nation", tableName));
```

Comment:
Do we expect exactly one file to be created? let's have an assertion on that

Reply:
Isn't `nation` to small. Would two writers be still used for a source table which only has 1 split (I assume nation has just one).
Maybe use UNION of a couple of NATION tables as a source.

Reply:
this was just an example from issue description https://github.com/trinodb/trino/issues/9149 I removed that whole test